lib/Jsrt/JsrtDiag.cpp

```diff
@@ -537,6 +537,10 @@ CHAKRA_API JsTTDDiagWriteLog(_In_reads_(uriLength) const char* uri, _In_ size_t
     return JsErrorCategoryUsage;
 #else
     return ContextAPIWrapper_NoRecord<true>([&](Js::ScriptContext * scriptContext) -> JsErrorCode {
+        if (!scriptContext->GetThreadContext()->IsRuntimeInTTDMode() || !scriptContext->GetThreadContext()->TTDLog->CanWriteInnerLoopTrace())
+        {
+            return JsErrorDiagUnableToPerformAction;
```

Comment:
Are callers of this method expecting such an error code & do they need to do anything special to account for it?

Reply:
Right now the caller just ignores the return code. However, the lack of an error code here is not inline with the form of the other API's so I wanted to get it added for cleanliness when I was adding the check code.